pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandlerTest.java

```diff
@@ -0,0 +1,432 @@
+        new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME).setNoDictionaryColumns(_noDictionaryColumns)
+            .setFieldConfigList(fieldConfigs).build();
+    tableConfig.setFieldConfigList(fieldConfigs);
+
+    indexLoadingConfig = new IndexLoadingConfig(null, tableConfig);
+    fwdIndexHandler = new ForwardIndexHandler(_existingSegmentMetadata, indexLoadingConfig);
+    operationMap = fwdIndexHandler.computeOperation(_writer);
+    assertEquals(operationMap.size(), 1);
+    assertEquals(operationMap.get(config.getName()), ForwardIndexHandler.Operation.CHANGE_RAW_COMPRESSION_TYPE);
+
+    // TEST6: Change compression and add index. Change compressionType for more than 1 column.
+    fieldConfigs = new ArrayList<>(_tableConfig.getFieldConfigList());
+    FieldConfig config1 = fieldConfigs.remove(0);
+    FieldConfig config2 = fieldConfigs.remove(1);
+
+    FieldConfig newConfig1 = new FieldConfig(config1.getName(), FieldConfig.EncodingType.RAW, Collections.emptyList(),
+        FieldConfig.CompressionCodec.ZSTANDARD, null);
+    fieldConfigs.add(newConfig1);
+    FieldConfig newConfig2 = new FieldConfig(config2.getName(), FieldConfig.EncodingType.RAW, Collections.emptyList(),
+        FieldConfig.CompressionCodec.ZSTANDARD, null);
+    fieldConfigs.add(newConfig2);
+
+    tableConfig =
+        new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME).setNoDictionaryColumns(_noDictionaryColumns)
+            .setFieldConfigList(fieldConfigs).build();
+    tableConfig.setFieldConfigList(fieldConfigs);
+
+    indexLoadingConfig = new IndexLoadingConfig(null, tableConfig);
+    indexLoadingConfig.getTextIndexColumns().add(config1.getName());
+    indexLoadingConfig.getInvertedIndexColumns().add(config1.getName());
+    fwdIndexHandler = new ForwardIndexHandler(_existingSegmentMetadata, indexLoadingConfig);
+    operationMap = fwdIndexHandler.computeOperation(_writer);
+    assertEquals(operationMap.size(), 2);
+    assertEquals(operationMap.get(config1.getName()), ForwardIndexHandler.Operation.CHANGE_RAW_COMPRESSION_TYPE);
+    assertEquals(operationMap.get(config2.getName()), ForwardIndexHandler.Operation.CHANGE_RAW_COMPRESSION_TYPE);
+  }
+
+  @Test
+  public void testRewriteRawForwardIndexForSingleColumn()
+      throws Exception {
+    for (int i = 0; i < _noDictionaryColumns.size(); i++) {
+
+      // For every noDictionaryColumn, change the compressionType to all available types, one by one.
+      for (FieldConfig.CompressionCodec compressionType : _allCompressionTypes) {
+        List<FieldConfig> fieldConfigs = new ArrayList<>(_tableConfig.getFieldConfigList());
+        FieldConfig config = fieldConfigs.remove(i);
+        String columnName = config.getName();
+        FieldConfig.CompressionCodec newCompressionType = compressionType;
+        if (config.getCompressionCodec().equals(newCompressionType)) {
+          // Compression types match. No need to invoke forward index handler.
```

Comment:
I suggest invoking the handler in the test, let the handler detect it as a NO-OP and you should assert for it here that nothing changed

Reply:
Done.